page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataFactory.java

```diff
@@ -1,61 +1,99 @@
+import ru.sbtqa.tag.datajack.TestDataProvider;
 import ru.sbtqa.tag.datajack.exceptions.DataException;
 import ru.sbtqa.tag.qautils.properties.Props;
 
+import java.lang.reflect.InvocationTargetException;
+
+import static java.lang.String.format;
+
 public class DataFactory {
 
-    private static TestDataObject dataContainer;
+    private static TestDataProvider testDataProvider;
+    private static final String BASE_FQDN = "ru.sbtqa.tag.datajack.providers.";
+
+    private enum PROVIDERS {
+        JSON_DATA_PROVIDER("json.JsonDataProvider"),
+        PROPERTIES_DATA_PROVIDER("properties.PropertiesDataProvider"),
+        EXCEL_DATA_PROVIDER("excel.ExcelDataProvider"),
+        MONGO_DATA_PROVIDER("mongo.MongoDataProvider");
+
+        private String value;
+
+        PROVIDERS(String value) {
+            this.value = value;
+        }
+
+        String getName() {
+            return value.split("\\.")[1];
+        }
+
+    }
 
     private DataFactory() {
-        throw new IllegalStateException("Utility class");
     }
 
-    public static TestDataObject getInstance() throws DataException {
-        if (dataContainer == null) {
-            String initialCollection = Props.get("data.initial.collection", null);
+    public static TestDataProvider getDataProvider() throws DataException {
+        if (testDataProvider == null) {
+            String initialCollection = Props.get("data.initial.collection");
             String dataFolder = Props.get("data.folder");
-            String dataType = Props.get("data.type", "stash");
+            String dataType = Props.get("data.type");
 
             switch (dataType) {
                 case "json":
-                    dataContainer = new JsonDataObjectAdaptor(
+                    testDataProvider = initProvider(PROVIDERS.JSON_DATA_PROVIDER,
                             dataFolder,
                             initialCollection,
                             Props.get("data.extension", "json")
                     );
                     break;
                 case "properties":
-                    dataContainer = new PropertiesDataObjectAdaptor(
+                    testDataProvider = initProvider(PROVIDERS.PROPERTIES_DATA_PROVIDER,
                             dataFolder,
                             initialCollection,
                             Props.get("data.extension", "properties")
                     );
                     break;
                 case "excel":
-                    dataContainer = new ExcelDataObjectAdaptor(
+                    testDataProvider = initProvider(PROVIDERS.EXCEL_DATA_PROVIDER,
                             dataFolder,
                             initialCollection
                     );
                     break;
+                case "mongo":
+                    testDataProvider = initProvider(PROVIDERS.MONGO_DATA_PROVIDER,
+                            new MongoClient(new MongoClientURI(Props.get("data.uri"))).getDB("data.db"),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "getDB"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=89&issues=AWW6t3Uu_eEZPtUmV-5f&open=AWW6t3Uu_eEZPtUmV-5f)

[ISSUE_KEY]: AWW6t3Uu_eEZPtUmV-5f